src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java

```diff
@@ -97,20 +98,78 @@ private Group renderEntities() {
 				entityBox.setTranslateZ(entity.getCenter().getZ());
 
 				if (entity instanceof Player) {
-					entityBox.setMaterial(new PhongMaterial(Color.ORANGE));
+					entityBox.setMaterial(Style.PLAYER);
 				} else if (entity instanceof Coin) {
-					entityBox.setMaterial(new PhongMaterial(Color.GOLD));
+					entityBox.setMaterial(Style.COIN);
 				} else if (entity instanceof Log) {
-					entityBox.setMaterial(new PhongMaterial(Color.BROWN));
+					entityBox.setMaterial(Style.WOOD);
 				} else if (entity instanceof Pillar) {
-					entityBox.setMaterial(new PhongMaterial(Color.WHITE));
-				} else {
-					entityBox.setMaterial(new PhongMaterial(Color.GRAY));
+					entityBox.setMaterial(Style.PILLAR);
+				} else /*if (entity instanceof Fence) */ {
+					entityBox.setMaterial(Style.FENCE);
 				}
+
+				entityBox.setCache(true);
+				entityBox.setCacheHint(CacheHint.SPEED);
 				entities.getChildren().add(entityBox);
             }
 		}
 		return entities;
 	}
+
+	/**
+	 * This method is for producing track pieces.
+	 * This is a working version, however refactoring is needed.
+	 * @return Group with all the track parts
+	 */
+	private Group renderTrack() {
+		Group result = new Group();
+		for (int i = 0; i < 100; i++) {
+			for (int j = 0; j < 3; j++) {
```

Comment:
I suggest replacing the for-loop boundaries by `Track.LENGTH` and `Track.WIDTH`.
<layout,logic